framework/src/kernels/KernelBase.C

```diff
@@ -34,6 +34,7 @@ InputParameters validParams<KernelBase>()
   params += validParams<MaterialPropertyInterface>();
 
   params.addRequiredParam<NonlinearVariableName>("variable", "The name of the variable that this Kernel operates on");
+  params.addParam<bool>("calculate_nonlocal_variable_jacobian", false, "Whether or not this object requires calculating jacobian contribution corresponding to non-local variables");
```

Comment:
Is the thought here that sometimes a user might want to turn it off because it's expensive... even though it's needed?

If you don't think a user should be able to turn it off then we really should just try to "detect" when it's needed instead of making it a parameter.